src/jit.rs

```diff
@@ -820,40 +913,45 @@ impl<'a> JitMemory<'a> {
                     emit_jcc(self, 0x8e, target_pc);
                 },
                 ebpf::CALL_IMM   => {
-                    // Save registers for call before displacement
-                    emit_register_saving_before_call(self);
-
                     // For JIT, syscalls MUST be registered at compile time. They can be
                     // updated later, but not created after compiling (we need the address of the
                     // syscall function in the JIT-compiled program).
-                    let func_ptr = match syscalls.get(&(insn.imm as u32)) {
-                        Some(Syscall::Function(func)) => *func as *const u8,
+                    match syscalls.get(&(insn.imm as u32)) {
+                        Some(Syscall::Function(func)) => {
+                            emit_rust_call(self, *func as *const u8, &[
+                                (1, Value::Register(ARGUMENT_REGISTERS[1])),
```

Comment:
Would there ever be gaps in the first value?

Reply:
Both `REGISTER_MAP` and `ARGUMENT_REGISTERS` use 0 as return value (or pointer to store the return value in) of the calling convention.